queue/abstract.lua

```diff
@@ -129,8 +129,8 @@ end
 -- connection id == "conn_id".
 -- Throw an error on failure.
 local function check_task_on_taken(tube_id, task_id, conn_id)
```

Comment:
AFAIU, the idea is to use {tube_id, task_id} to search a task instead of {conn_id, tube_id, task_id}, while both indices are unique. It would be great to leave some explanation, why it is faster.

Reply:
I don't know "why". I write benchmarks and try several variants. AFAIK this is a valid way to choose a more optimal solution.